physics/continuous_trajectory_body.hpp

```diff
@@ -452,16 +454,30 @@ template<typename Frame>
 typename ContinuousTrajectory<Frame>::InstantPolynomialPairs::const_iterator
 ContinuousTrajectory<Frame>::FindPolynomialForInstant(
     Instant const& time) const {
-  // Need to use |lower_bound|, not |upper_bound|, because it allows
-  // heterogeneous arguments.  This returns the first polynomial |p| such that
-  // |time <= p.t_max()|.
-  auto const it = std::lower_bound(
-                      polynomials_.begin(), polynomials_.end(), time,
-                      [](InstantPolynomialPair const& left,
-                         Instant const& right) {
-                        return left.t_max < right;
-                      });
-  return it;
+  // This returns the first polynomial |p| such that |time <= p.t_max|.
+  {
+    auto const begin = polynomials_.begin();
+    auto const it = begin + last_accessed_polynomial_;
+    int const last_accessed_polynomial = last_accessed_polynomial_;
+    if (it != polynomials_.end() && time <= it->t_max &&
+        (it == begin || std::prev(it)->t_max < time)) {
+      return it;
+    }
+  }
+  {
+    // Need to use |lower_bound|, not |upper_bound|, because it allows
+    // heterogeneous arguments.
```

Comment:
I don't understand this comment (see [`lower_bound`](http://en.cppreference.com/w/cpp/algorithm/lower_bound), [`upper_bound`](http://en.cppreference.com/w/cpp/algorithm/upper_bound)).

Reply:
Removed.